pydss/flapper.py

```diff
@@ -70,103 +61,82 @@ class Bid :
       uint256 bid
     );
     """
-    vat = None;
-    gem = None
+
 
     def __init__(self, vat, gem,msg) :
+        self.wards = dict()
         self.message = msg
         self.wards[msg.sender] = 1;
         self.vat = VatLike(vat);
         self.gem = GemLike(gem);
         self.live = 1;
+        self.bids = dict()
+        self.ONE = Wad.from_number(1.00)
+        self.beg = Wad.from_number(1.05)
+        # 3 hours bidduration[seconds]
+        self.ttl = 3 * 60 * 60;
+        # 2 days total auction length  [seconds]
+        self.tau = 2 * 24 * 60 * 60
+        self.kicks = 0;
+        self.live = 0
+        self.vat = vat
+        self.gem =gem
+
 
-    def add(self, x, y):
-        z = x+y
-        if z > x:
-            return z
-        return -1
-
-    def mul(self, x, y):
-        z = x * y
-        if (y == 0 or (z // y == x)):
-            return z
-        return -1
     def kick(self, lot, bid) :
-        if (self.live != 1):
-            return "Flapper/not-live"
-        if (self.kicks < -1):
-            return "Flapper/overflow";
-        sid = id(self)
-        sid += self.kicks;
-
-        self.bids[sid].bid = bid;
-        self.bids[sid].lot = lot;
-        self.bids[sid].guy = self.message.sender;
-        self.bids[sid].end = self.add(self.now, self.tau);
+        require(self.live == 1,"Flapper/not-live")
+
+        require(self.kicks < -1,"Flapper/overflow")
+
+        self.kicks+=1
+        self.bid_id += self.kicks;
```

Comment:
`bid_id` shouldn't be a field on `self`, it just pertains to the bid being kicked in the function. this should probably look like:
```suggestion
        bid_id = self.kicks;
```
and all future uses of `self.bid_id` should just be `bid_id`